garage/misc/git_handler.py

```diff
@@ -0,0 +1,336 @@
+"""Perform different git tasks while running experiments."""
```

Comment:
This is an immense amount of code for a small feature.

Can you explain what use cases this implements?

Reply:
Currently, run an experiment in a target commit, but all the code in this class is used for that purpose, except for the creation and removal of branches and tags, but those are required for my tests.
Also, the methods in this class will be used to implement two more features:
- Add the SHA to the experiment log.
- Create a new commit and push it to the remote